mis_builder_budget_contrib/models/mis_budget_by_account_item.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2022 Camptocamp SA (https://www.camptocamp.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class MisBudgetByAccountItem(models.Model):
+
+    _inherit = "mis.budget.by.account.item"
+    _order = "budget_id, date_from, account_id, product_id"
+
+    product_id = fields.Many2one(
+        comodel_name="product.product",
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
+    )
+
+    def _prepare_overlap_domain(self):
+        domain = super()._prepare_overlap_domain()
+        domain.extend([("product_id", "=", self.product_id.id)])
+        return domain
```

Comment:
Can you please add a comment explaining the expected behavior if `product_id` is not defined on `self` or, viceversa, why we can expect it to be always properly filled?

Reply:
If the `product_id` is not defined, it was detecting overlapping with another empty line (`product_id` not defined <=> `product_id = product.product()` <=> `product_id.id = product.product().id = False`)

But thanks to your remark I saw another issue : it was not counting overlapping for a line with no product (including all products) and a line with one product.

Changed accordingly, so I answer you here instead of in comments